src/libra_py/cube_file_process.py

```diff
@@ -0,0 +1,255 @@
+    """
+	This function reads the wavefunction from a cube file and calculate 
+	the grid volum using the X-, Y- and Z-axis of the volumetric region
+	which are placed in the 4th, 5th and 6th line of the cube file structure
+	
+        Args:
+        
+	    filename (string): The name of the .cube file to read.
+		
+        Returns:
+	
+	    dv (float): The grid volume in Bohr^3.
+		
+        Note:
+        
+	    It uses the formula of calculating the parallelpiped volume:
+            https://en.wikipedia.org/wiki/Parallelepiped
+        	
+    """
+	
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+    
+    # We use the 3rd, 4th and 5th row in the lines to obtain
+    # the vectors of the parallelpiped into a numpy array (V).
+    V = np.array([[float(lines[3].split()[1]),float(lines[3].split()[2]),float(lines[3].split()[3])],\
+                 [float(lines[4].split()[1]),float(lines[4].split()[2]),float(lines[4].split()[3])],\
+                 [float(lines[5].split()[1]),float(lines[5].split()[2]),float(lines[5].split()[3])]\
+                 ])
+    # Then we calculate the determinant of the a to obtain the volume.
+    dv = np.absolute(np.linalg.det(V))
+	
+    
+    return dv 
+
+
+
+
+def read_volumetric_data(filename: str):
+    """
+    This function reads the volumetric data in a format used for plotting
+    of the data. The difference between 'read_cube' function is that it will
+    show the data in a 3D array which has the shape as the number of grid points
+    for each of the X-, Y-, and Z-axis in the 4th to 6th line of the cube file.
+	
+    This function will return the grid points in each axis, the coordinates of the
+    structure and the spacing vector which is used to plot the isosurfaces of the 
+    molecular orbitals.
+	
+    Args:
+	
+        filename (string): The name of the .cube file.
+		
+    Returns:
+	
+        Coordinates (2D numpy array): The coordinates of the molecule structure in
+                                      the same format shown in the .cube file.
+		
+           x, y, z (3D numpy array): Containing the grid points for each of the X-,
+                                     Y-, and Z- axis.
+		
+           v (3D numpy array): The volumetric data in a 3D numpy array format.
+		
+           spacing (numpy 1D array): The spacing vector used for plotting the isosurfaces.
+
+    """
+
+    f = open(filename,'r')
+    lines = f.readlines()
+    f.close()
+	
+	# The number of atoms in the 3rd line
+    natoms = int(lines[2].split()[0])
+	
+	# The number of voxels defined for each axis
+    vx = int(lines[3].split()[0])
+    vy = int(lines[4].split()[0])
+    vz = int(lines[5].split()[0])
+	
+	# The spacing vector. This will be used in the 'marching_cubes_lewiner'
+	# function to define the vertices and faces for 'plot_trisurf' function.
+    spacing = []
+    spacing.append([float(lines[3].split()[1])+float(lines[4].split()[1])+float(lines[5].split()[1]),\

```

Comment:
what is the meaning of those "float(lines[3].split()[1])" etc. ? Please first introduce some variables with more transparent purpose, set them up from the "lines" and then use in the following operations, so that one knows what is going on

Same in the code below